src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1588,6 +1591,10 @@ private BoundNode GetBoundLambdaOrQuery(CSharpSyntaxNode lambdaOrQuery)
             {
                 BoundNode boundOuterExpression = this.Bind(incrementalBinder, nodeToBind, _ignoredDiagnostics);
                 nodes = GuardedAddBoundTreeAndGetBoundNodeFromMap(lambdaOrQuery, boundOuterExpression);
+
+                // If we're here, nullable analysis must be off, or we're in an error scenario and didn't find any
```

Comment:
>// If we're here, nullable analysis must be off, or we're in an error scenario and didn't find any [](start = 16, length = 98)

I think we should have a prototype comment here instead of an assert. We need to make a decision what to do about code that we bind in error recovery situations and then adjust this place accordingly. If we can ever reach this code with ```!nodes.IsDefaultOrEmpty```, the I have no doubt it can be reached with ```EnableNullableAnalysis``` at the same time. So the assert below is simply indicating a possible test gap rather than the fact that it is safe to do nothing here.